### PR TITLE
Implement Ask and Validate methods for PackageAppOpts

### PR DESCRIPTION
**Pull request by efekarakus**
Retrieve the inputs for https://github.com/aws/amazon-ecs-cli-v2/issues/229

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/app_package_test.go [hunk 1]
```diff
@@ -2,3 +2,293 @@
...
+	}{
+		"invalid workspace": {
+			expectWS: func(m *mocks.MockWorkspace) {
+				m.EXPECT().AppNames().Times(0)
+			},
+			expectEnvStore: func(m *mocks.MockEnvironmentStore) {
+				m.EXPECT().GetEnvironment(gomock.Any(), gomock.Any()).Times(0)
+			},
+
+			wantedErrorS: "could not find a project attached to this workspace, please run `project init` first",
```
Comment (SoManyHs) [inline internal/pkg/cli/app_package_test.go]: Nit: For maintainability, should we use the `errNoWorkspace` variable defined in https://github.com/aws/amazon-ecs-cli-v2/pull/260/files#diff-a0a7114926f9397ff0487c11217501deR14?

File: internal/pkg/cli/app.go [hunk 2]
```diff
@@ -4,11 +4,15 @@
...
 
 import (
+	"errors"
+
 	"github.com/aws/amazon-ecs-cli-v2/cmd/archer/template"
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/cli/group"
 	"github.com/spf13/cobra"
 )
 
+var errNoWorkspace = errors.New("could not find a project attached to this workspace, please run `project init` first")
```
Comment (SoManyHs) [inline internal/pkg/cli/app.go]: Should this be called `errNoProject` instead? 🤔

File: internal/pkg/cli/app_package.go [hunk 3]
```diff
@@ -40,11 +49,54 @@ func NewPackageAppOpts() *PackageAppOpts {
...
 func (opts *PackageAppOpts) Ask() error {
+	if opts.AppName == "" {
+		names, err := opts.listAppNames()
+		if err != nil {
+			return err
+		}
+		if len(names) == 0 {
+			return errors.New("there are no applications in the workspace, run `archer init` first")
+		}
+		app, err := opts.prompt.SelectOne("Which application's CloudFormation template would you like to generate?", "", names)
```
Comment (SoManyHs) [inline internal/pkg/cli/app_package.go]: Nit: I tend to dislike using possessives in this way. Maybe "Which application would you like to generate a CFN template for?"?

File: internal/pkg/cli/app_package.go [hunk 4]
```diff
@@ -40,11 +49,54 @@ func NewPackageAppOpts() *PackageAppOpts {
...
+			return fmt.Errorf("prompt application name: %w", err)
+		}
+		opts.AppName = app
+	}
+	if opts.EnvName == "" {
+		names, err := opts.listEnvNames()
+		if err != nil {
+			return err
+		}
+		env, err := opts.prompt.SelectOne("Which environment's configuration would you like to use for your stack's parameters?", "", names)
```
Comment (SoManyHs) [inline internal/pkg/cli/app_package.go]: This prompt also a bit weirdly worded. What's an environment configuration?

File: internal/pkg/cli/app_package.go [hunk 5]
```diff
@@ -53,6 +105,36 @@ func (opts *PackageAppOpts) Execute() error {
 	return nil
 }
 
+func (opts *PackageAppOpts) listAppNames() ([]string, error) {
+	names, err := opts.ws.AppNames()
+	if err != nil {
+		return nil, fmt.Errorf("list applications in workspace: %w", err)
```
Comment (SoManyHs) [inline internal/pkg/cli/app_package.go]: Wasn't there previous PR where the verb tense of the errors was confusing? Did that get resolved?

File: internal/pkg/cli/app_package_test.go [hunk 6]
```diff
@@ -2,3 +2,293 @@
...
+					{
+						Name: "test",
+					},
+					{
+						Name: "prod",
+					},
+				}, nil)
+			},
+			expectPrompt: func(m *climocks.Mockprompter) {
+				m.EXPECT().SelectOne("Which environment's configuration would you like to use for your stack's parameters?", gomock.Any(), []string{"test", "prod"}).Return("test", nil)
```
Comment (SoManyHs) [inline internal/pkg/cli/app_package_test.go]: Again might be good to constantize prompts to make tests more maintainable

File: internal/pkg/cli/app.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/cli/app.go]: That makes sense, changed to `errNoProjectInWorkspace`!

File: internal/pkg/cli/app_package.go (hunk 3 above)
Comment (efekarakus) [reply internal/pkg/cli/app_package.go]: :+1: that is better! Updated.

File: internal/pkg/cli/app_package.go (hunk 4 above)
Comment (efekarakus) [reply internal/pkg/cli/app_package.go]: Ah yeah, I meant if they want to use environment overrides from their manifest we need to know their environment's name.

Changed to `"Which environment would you like to create this stack for?"`

File: internal/pkg/cli/app_package.go (hunk 5 above)
Comment (efekarakus) [reply internal/pkg/cli/app_package.go]: See previous comment 😝 https://github.com/aws/amazon-ecs-cli-v2/pull/250#discussion_r341383402

File: internal/pkg/cli/app_package_test.go (hunk 6 above)
Comment (efekarakus) [reply internal/pkg/cli/app_package_test.go]: :+1: good idea! Moved the prompts to constants

**Pull request merged**